shared/src/commonMain/kotlin/com.macaosoftware.sdui.app/marketplace/amadeus/auth/AuthViewModel.kt

```diff
@@ -26,12 +37,85 @@ class AuthViewModel(
 
     }
 
-    fun login() {
+    suspend fun loginWithEmailAndLink(email: String, link: String) {
+        authPlugin.loginEmailAndLink(
+            LoginRequestForEmailWithLink(
```

Comment:
Shouldn't be here but in the plugin, add this function to the AuthPlugin interface